test/unit/question_base_test.rb

```diff
@@ -90,6 +90,19 @@ class QuestionBaseTest < ActiveSupport::TestCase
         assert_equal [:done], @question.permitted_next_nodes
       end
     end
+
+    context 'next_node called with block but not permitted option specified' do
+      should 'return nodes returned via syntactic sugar methods' do
+        @question.next_node do |response|
+          if response == 'yes'
+            outcome :done
+          else
+            question :another_question
+          end
+        end
+        assert_equal [:done, :another_question], @question.permitted_next_nodes
+      end
+    end
   end
```

Comment:
In the context description, when I read "not permitted option specified" I'm looking for the 'not permitted' option, but I guess it means without a permitted option.

In the should description, I'm not sure I'm aware of what syntactic sugar methods relates to. I think a more obvious expectation would be that it expects to return a list of nodes from the outcomes and questions defined within the block.